llvm/test/tools/sycl-post-link/device-requirements/device-requirements.ll

```diff
@@ -0,0 +1,207 @@
+; Original code:
+; #include <sycl/sycl.hpp>
+; [[__sycl_detail__::__uses_aspects__(sycl::aspect::fp64, sycl::aspect::cpu)]] void foo() {}
+; [[__sycl_detail__::__uses_aspects__(sycl::aspect::queue_profiling, sycl::aspect::cpu, sycl::aspect::image)]] void bar() {
+;   foo();
+; }
+; int main() {
+;   sycl::queue q;
+;   q.submit([&](sycl::handler &cgh) {
+;     cgh.single_task([=]() { bar(); });
+;   });
+; }
+
+; RUN: sycl-post-link -split=auto -o %s -o %t.files.table
+; RUN: FileCheck %s -input-file=%t.files_2.prop --check-prefix CHECK-PROP
+
+; CHECK-PROP: [SYCL/device requirements]
+; CHECK-PROP-NEXT: aspects=2|ACAAAAAAAAQAAAAAGAAAAkAAAAADAAAA
+
+source_filename = "llvm-link"
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+%class.anon = type { i8 }
+
+$_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_EUlvE_ = comdat any
+
+@__spirv_BuiltInWorkgroupId = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+@__spirv_BuiltInGlobalLinearId = external dso_local local_unnamed_addr addrspace(1) constant i64, align 8
+@__spirv_BuiltInWorkgroupSize = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+
+; Function Attrs: convergent mustprogress noinline norecurse optnone
+define weak_odr dso_local spir_kernel void @_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_EUlvE_() #0 comdat !kernel_arg_buffer_location !6 {
+entry:
+  call void @__itt_offload_wi_start_wrapper()
```

Comment:
Could you please re-generate this test using `-fno-sycl-instrument-device-code` flag? this should reduce the amount of LLVM IR significantly, making the test simpler

Reply:
Wow, good prompt, thanks.